### PR TITLE
builtin: Fix hex format of ascii

### PR DESCRIPTION
**Pull request by corona10**
Some range of Unicode characters should be specially handled

### CODE REVIEW

**Comment by corona10** [comment]
@windelbouwman 

PTAL

**Comment by corona10** [comment]
@windelbouwman

PTAL

**Comment by windelbouwman** [comment]
Hi @corona10 , what does PTA and PTAL mean?

File: vm/src/builtins.rs [hunk 1]
```diff
@@ -64,10 +64,13 @@ fn builtin_ascii(obj: PyObjectRef, vm: &VirtualMachine) -> PyResult<String> {
     let mut ascii = String::new();
     for c in repr.value.chars() {
         if c.is_ascii() {
-            ascii.push(c)
+            ascii.push(c);
```
Comment (windelbouwman) [inline vm/src/builtins.rs]: The trailing semi column `;` might not be required here.

File: vm/src/builtins.rs [hunk 2]
```diff
@@ -64,10 +64,13 @@ fn builtin_ascii(obj: PyObjectRef, vm: &VirtualMachine) -> PyResult<String> {
...
+        } else if (c as i64) < 0x10000 {
```
Comment (windelbouwman) [inline vm/src/builtins.rs]: Consider converting only once to `i64`, like this:
```rust
if c.is_ascii() {
    ...
} else {
    let c = c as i64;
    let hex = if c < 0x10000 {
        format!("\\u{:04x}", c)
    } else {
        format!("\\U{:08x}", c)    
    };
    ascii.push_str(&hex);
}

Comment (corona10) [reply vm/src/builtins.rs]: Oh, this is great! Thanks!

**Comment by corona10** [comment]
@windelbouwman 

PTAL stands for 'Please take a look'  :) (PTA was my typo.. sorry)
I 've updated your review, Thanks for the review with cool tips!

**Pull request merged**